source/PlayerInfoPanel.cpp

```diff
@@ -138,19 +138,19 @@ namespace {
 	}
 
 	// Reverses the argument order of the given comparator function.
-	InfoPanelState::ShipComparator &GetReverseCompareFrom(InfoPanelState::ShipComparator &f)
+	InfoPanelState::ShipComparator &GetReverseCompareFrom(InfoPanelState::ShipComparator f)
```

Comment:
This was a pointer to ShipComparator, so I expected the parameter to be `*f`, but both `f` as well as `*f` seems to work. Does C++ perform some automated conversion magic here? (And we are not accidentally calling by value?)

Reply:
Isn't that what he is doing, removing the comparators?

Reply:
Comparatir is a shared_ptr

Reply:
> This was a pointer to ShipComparator, so I expected the parameter to be `*f`, but both `f` as well as `*f` seems to work (I tested both). Does C++ perform some automated conversion magic here? (And we are not accidentally calling by value?)

I *think* I am removing a call-by-value and just passing a function pointer instead. But it's a bit hard to follow (then again, it is C++, so…)

I am pretty sure the code as suggested in this PR is just passing around function pointers and not doing anything trippy.